src/csharp/Microsoft.Spark/Interop/SparkEnvironment.cs

```diff
@@ -14,10 +14,30 @@ namespace Microsoft.Spark.Interop
     /// </summary>
     internal static class SparkEnvironment
     {
-        private static readonly Lazy<Version> s_sparkVersion = new Lazy<Version>(
-            () => new Version((string)JvmBridge.CallStaticJavaMethod(
+        private static readonly ILoggerService s_logger =
+            LoggerServiceFactory.GetLogger(typeof(SparkEnvironment));
+
+        private static Version GetSparkVersion()
+        {
+            var sparkVersion = new Version((string)JvmBridge.CallStaticJavaMethod(
                 "org.apache.spark.deploy.dotnet.DotnetRunner",
-                "SPARK_VERSION")));
+                "SPARK_VERSION"));
+
+            string sparkVersionOverride =
+                Environment.GetEnvironmentVariable("SPARK_VERSION_OVERRIDE");
+            if (!string.IsNullOrEmpty(sparkVersionOverride))
+            {
+                s_logger.LogInfo(
+                    $"Overriding the Spark version from '{sparkVersion}' " +
+                    $"to '{sparkVersionOverride}'.");
+                sparkVersion = new Version(sparkVersionOverride);
+            }
+
+            return sparkVersion;
+        }
```

Comment:
If the `SPARK_VERSION_OVERRIDE` is defined then return that value, otherwise call and return the value of `org.apache.spark.deploy.dotnet.DotnetRunner.SPARK_VERSION`

Reply:
I prefer to print the original version if the version is overridden for debugging.

Reply:
I prefer to print the original version if the version is overridden for debugging.

Reply:
Sure that's fair.

Reply:
Sounds good.